### DISCUSSION TITLE
Error: Unable to create image, image not found: unable to get digest: Got bad response from registry: 404 Not Found

### DISCUSSION THREAD

**Discussion by sdgoldstein** [kreuzwerker/terraform-provider-docker#557] [Q&A]
I'm running into this issue after the docker image is pushed to the registry:

"Error: Unable to create image, image not found: unable to get digest: Got bad response from registry: 404 Not Found"

I've checked:
1.  Docker debug logs - nothing relevant
2. Terraform Trace Logs - nothing much here either
3. Ran "docker images {url}" and this correctly returns the image I'm expecting, confirming that it was pushed

My terraform code is fairly simple:

```
provider "docker" {
  registry_auth {
    address  = aws_ecr_repository.ecr_repo.repository_url
    username = data.aws_ecr_authorization_token.ecr_auth_token.user_name
    password = data.aws_ecr_authorization_token.ecr_auth_token.password
  }
}

resource "docker_image" "app_image" {
  name = "${aws_ecr_repository.ecr_repo.repository_url}:latest"
  build {
    context = "${path.cwd}/docker"
  }
}

resource "docker_registry_image" "app_image" {
  name = docker_image.app_image.name
}
```

And the version I'm using:

```
docker = {
      source = "kreuzwerker/docker"
      version = ">= 3.0.2"
}
```

Any thoughts on why this might be happening?  Is there any way that I can see what values the provider is passing to getImageDigest(...)?

Thanks!

**Comment by mkgeneral** [comment]
I am getting the same error message in 2 scenarios: 
 - the first one is described above and 
 - the second scenario is when an image is being read from a private registry using `data "docker_registry_image"`. 
 
It fails whenever either image itself or its digest is read  by `resource_docker_registry_image_funcs.getImageDigestWithFallback(...)` or `data_source_docker_registry_image.dataSourceDockerRegistryImageRead(...)`.

Both functions call ` getImageDigest(...)` as noted above.

**Comment by legioner0** [comment]
I've got same error on private registry due to my own mistake - `registry_auth.address` contains value with path (`cr.example.com/some_path`) instead of host (`cr.example.com`). Provider doesn't check value and as a result try to get image digest from wrong url `cr.example.com/some_path/v2/some_path/<image>/manifest/<tag>`.
